src/crimson/osd/main.cc

```diff
@@ -94,7 +94,8 @@ int main(int argc, char* argv[])
       }).then([&osd, mkfs = config.count("mkfs")] {
         if (mkfs) {
           return osd.invoke_on(0, &OSD::mkfs,
-                               local_conf().get_val<uuid_d>("fsid"));
+                               local_conf().get_val<uuid_d>("fsid"))
```

Comment:
see: https://github.com/scylladb/seastar/wiki/Seastar-Tutorial,  It said as follows:  
When the future returned by it resolves the app will shut down, stopping event loops on all cpus, and app.run() will return.
There's also app.run_deprecated() variant, still in use by some code, which differs from run() in that the application doesn't exit when the callback returns, but it has to be explicitly stopped by calling engine_exit(). It should be used instead of the regular C exit(), to allow for proper cleanups when necessary. 

it is said app.run_deprecated still in use by some code, means it is deprecated,  only when for proper cleanups when necessary,  If we can deal the stop confliction, we'd better not use run_deprecated.  I think we can arrange the code to avoid this.

Reply:
i am not against using `run()` as long as we have a fix. please send a patch to address this issue.